pkg/apis/extensions/types.go

```diff
@@ -809,3 +810,75 @@ type PodSecurityPolicyList struct {
 
 	Items []PodSecurityPolicy `json:"items"`
 }
+
+// +genclient=true
+
+// Template contains one or more partially complete configs with substitution parameters.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	Spec TemplateSpec `json:"spec"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
+	// Template to Config transformation.
+	Parameters []Parameter `json:"parameters,omitempty"`
+
+	// Required: A yaml config list of resources to create
+	Objects []runtime.RawExtension `json:"objects"`
+}
```

Comment:
Yes.  Apologies for not looping you into this directly.  I chatted with @lavalamp and @bgrant0607 a bit about this and decided to defer ObjectLabels support.

There were 2 main concerns with how ObjectLabels should behave: 
- Objects such as Deployment use label defaulting to apply pod-labels to the Deployment if the Deployment has no labels.  ObjectLabels would break the defaulting because the Deployment labels would be non-empty
- What is the correct thing to do with recursion?  Should the labels be applied to the PodTemplate object in a Deployment?  What about elements of Lists?

We may want to instead consider supporting map[string]string as a substitution type so that the user can be more explicit about how they want things wired together.

Since it will be easier to add some support for object labels in the future than it will be to change the behavior if we already support them, I think we should hold off on this until we know exactly what we want.